src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/SourceGeneratorAttributeAnalyzerFix.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = nameof(SourceGeneratorAttributeAnalyzerFix))]
+    [Shared]
+    public sealed class SourceGeneratorAttributeAnalyzerFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(
+            DiagnosticIds.MissingSourceGeneratorAttributeId);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span);
+
+            if (node is null)
+            {
+                return;
+            }
+
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                AddFix(
+                    string.Format(CodeAnalysisDiagnosticsResources.AddGeneratorAttribute_1, LanguageNames.CSharp),
+                    context, document, node, diagnostic, LanguageNames.CSharp);
+
+                AddFix(
+                    string.Format(CodeAnalysisDiagnosticsResources.AddGeneratorAttribute_1, LanguageNames.VisualBasic),
+                    context, document, node, diagnostic, LanguageNames.VisualBasic);
+
+                AddFix(
+                    string.Format(CodeAnalysisDiagnosticsResources.AddGeneratorAttribute_2, LanguageNames.CSharp, LanguageNames.VisualBasic),
+                    context, document, node, diagnostic, LanguageNames.CSharp, LanguageNames.VisualBasic);
+            }
+        }
+
+        private static void AddFix(string title, CodeFixContext context, Document document, SyntaxNode node, Diagnostic diagnostic, params string[] languageNames)
+        {
+            var codeAction = CodeAction.Create(
+                title,
+                (cancellationToken) => FixDocumentAsync(document, node, languageNames, cancellationToken),
+                equivalenceKey: nameof(SourceGeneratorAttributeAnalyzerFix));
+
+            context.RegisterCodeFix(codeAction, diagnostic);
+        }
+
+        public override FixAllProvider? GetFixAllProvider() => null;
```

Comment:
First time I see a null value returned from `GetFixAllProvider`, I'm curious to know what does it mean and why we're not returning `WellKnownFixAllProviders.BatchFixer`

Reply:
Let me add it and some tests for fix all